File: 500/594 - Longest Harmonious Subsequence.java
import java.util.HashMap;

class Solution {
    public int findLHS(int[] nums) {
        HashMap<Integer, Integer> hm = new HashMap<Integer, Integer>();
        int ret = 0;
        
        for (int num : nums) {
            if (hm.containsKey(num))
                hm.replace(num, hm.get(num) + 1);
            else
                hm.put(num, 1);
        }
        
        for (HashMap.Entry<Integer, Integer> e : hm.entrySet()) {
            if (hm.containsKey(e.getKey() + 1))
                ret = Math.max(ret, e.getValue() + hm.get(e.getKey() + 1));
        }
        
        return ret;
    }
}

File: csharp/594.longest-harmonious-subsequence.cs
/*
 * @lc app=leetcode id=594 lang=csharp
 *
 * [594] Longest Harmonious Subsequence
 */

using System.Collections.Generic;

// @lc code=start
public class Solution {
    public int FindLHS(int[] nums) {
        int ret = 0;
        IDictionary<int, int> count = new Dictionary<int, int>();
        foreach (int num in nums) {
            int c = 0;
            count.TryGetValue(num, out c);
            count[num] = c + 1;
        }
        foreach (var pair in count) {
            int num = pair.Key;
            int c;
            if (count.TryGetValue(num + 1, out c)) {
                ret = Math.Max(ret, c + pair.Value);
            }
        }
        return ret;
    }
}
// @lc code=end

